compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -558,6 +558,29 @@ fn transcribe_metavar_expr<'a>(
         span
     };
     match *expr {
+        MetaVarExpr::Concat((lhs, lhs_sign), (rhs, rhs_sign)) => {
+            let string = |ident: Ident, has_sign: bool| {
+                if !has_sign {
+                    return ident.to_string();
+                }
+                let mrni = MacroRulesNormalizedIdent::new(ident);
+                if let Some(nm) = lookup_cur_matched(mrni, interp, &repeats)
+                    && let MatchedNonterminal(nt) = nm
+                    && let Nonterminal::NtIdent(nt_ident, _) = &nt.0
+                {
+                    nt_ident.to_string()
+                } else {
+                    ident.to_string()
+                }
+            };
+            let symbol_span = lhs.span.to(rhs.span);
+            let mut symbol_string = string(lhs, lhs_sign);
+            symbol_string.push_str(&string(rhs, rhs_sign));
```

Comment:
can just be `string(lhs, lhs_sign) + &string(rhs, rhs_sign)` :)

Reply:
Done